src/gui/Connected.qml

```diff
@@ -701,6 +739,40 @@ Item {
                             anchors.verticalCenter: parent.verticalCenter
                         }
                     }
+
+                    Button {
+                        id: refreshButton
+                        text: "Refresh Devices"
+                        anchors.right: closePopupButton.left
+                        anchors.rightMargin: rightMargin * virtualstudio.uiScale
+                        anchors.bottomMargin: rightMargin * virtualstudio.uiScale
+                        anchors.bottom: parent.bottom
+                        width: 150 * virtualstudio.uiScale; height: 30 * virtualstudio.uiScale
+
+                        palette.buttonText: textColour
+                        background: Rectangle {
+                            radius: 6 * virtualstudio.uiScale
+                            color: refreshButton.down ? buttonPressedColour : (refreshButton.hovered ? buttonHoverColour : buttonColour)
+                            border.width: 1
+                            border.color: refreshButton.down ? buttonPressedStroke : (refreshButton.hovered ? buttonHoverStroke : buttonStroke)
```

Comment:
I think some of these colors might not be defined. Got some terminal warnings from here.

Reply:
> I just tried it on my Windows laptop -it doesn't crash but it does freeze up unrecoverably. The terminal says "Stopping JackTrip..." and then "Peer Stopped", at which point all I can do is close the frozen window and restart the app. No plugging or unplugging devices or anything. Also couldn't try it with ASIO because I think ASIO might be broken on my Windows laptop.

Same. By "crash" I mean it freezes whenever I click it. Not doing any changes to devices.

Reply:
Terminal Dump:

```
INPUT DEVICE:
Audio Device  [WASAPI - 2] : Microphone (Realtek(R) Audio)
  Output Channels : 0
  Input Channels  : 2
  Supported Sampling Rates: 4000 5512 8000 9600 11025 16000 22050 32000 44100 48000 88200 96000 176400 192000
  --Default Input Device--
  --Probed Successful--
---------------------------------------------------------
OUTPUT DEVICE:
Audio Device  [WASAPI - 0] : Headphones (Realtek(R) Audio)
  Output Channels : 2
  Input Channels  : 0
  Supported Sampling Rates: 4000 5512 8000 9600 11025 16000 22050 32000 44100 48000 88200 96000 176400 192000
  --Default Output Device--
  --Probed Successful--
---------------------------------------------------------
The Sampling Rate is: 48000
---------------------------------------------------------
The Audio Buffer Size is: 128 samples
                      or: 512 bytes
---------------------------------------------------------
The Number of Channels is: 2
---------------------------------------------------------
Using UDP Protocol
---------------------------------------------------------
Using experimental buffer strategy 4-- Regulator with PLC (worker=false)
Server Address set to: 54.151.25.85 Port: 61002
---------------------------------------------------------
UDP Socket Receiving in Port: 4052
---------------------------------------------------------
Waiting for Peer...
Initializing Faust plugins (have 4) at sampling rate 48000
Received Connection from Peer!
PLC is in auto mode and has been set with 5 ms headroom
Received connection
UDP waiting too long (more than 30ms) for 54.151.25.85...
UDP waiting too long (more than 30ms) for 54.151.25.85...
UDP waiting too long (more than 30ms) for 54.151.25.85...
qrc:/vs/Connected.qml:757: ReferenceError: buttonPressedStroke is not defined
qrc:/vs/Connected.qml:755: ReferenceError: buttonPressedColour is not defined
Cannot connect to named pipe after wait = \\.\pipe\server_jack_default_0 err = 2
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

RtApiAsio::getDeviceInfo: error (Hardware input or output is not present or available.) initializing driver (JackRouter).

jack_deactivate called with a NULL client
jack_deactivate called with a NULL client
jack_client_close called with a NULL client
Cannot connect to named pipe after wait = \\.\pipe\server_jack_default_0 err = 2
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

RtApiAsio::getDeviceInfo: error (Hardware input or output is not present or available.) initializing driver (JackRouter).

jack_deactivate called with a NULL client
jack_deactivate called with a NULL client
jack_client_close called with a NULL client
perform reconnect things
Stopping JackTrip...
Peer Stopped

```